lib/config/helper.go

```diff
@@ -0,0 +1,373 @@
+			if err != nil {
+				errorList.PushBack(err.Error())
+			}
+		default:
+			continue
+		}
+	}
+	return getSingleErrorFromList(errorList)
+}
+
+func validateRootCACertificate(certBytes []byte, svc string) error {
+	rootCA, err := x509.ParseCertificate(certBytes)
+	if err != nil {
+		return fmt.Errorf("failed to parse root_ca for %s: %v", svc, err)
+	}
+	if time.Now().After(rootCA.NotAfter) {
+		return fmt.Errorf("root_ca for %s has expired", svc)
+	}
+	return nil
+}
+
+func validatePostgresqlCerts(postgresqlSettings *ConfigSettings) error {
+	if len(strings.TrimSpace(postgresqlSettings.RootCA)) < 1 ||
+		len(strings.TrimSpace(postgresqlSettings.PrivateKey)) < 1 ||
+		len(strings.TrimSpace(postgresqlSettings.PublicKey)) < 1 {
+		return fmt.Errorf("postgresql root_ca and/or public_key and/or private_key are missing. Otherwise set enable_custom_certs to false")
+	}
+
+	return checkCertValid([]keydetails{
+		{key: postgresqlSettings.RootCA, certtype: ROOT_CA, svc: POSTGRESQL},
+		{key: postgresqlSettings.PrivateKey, certtype: PRIVATE_KEY, svc: POSTGRESQL},
+		{key: postgresqlSettings.PublicKey, certtype: PUBLIC_KEY, svc: POSTGRESQL},
+	})
+}
+
+func validateOpensearchCerts(opensearchSettings *ConfigOpensearchSettings) error {
+	if len(strings.TrimSpace(opensearchSettings.RootCA)) < 1 ||
+		len(strings.TrimSpace(opensearchSettings.AdminKey)) < 1 ||
+		len(strings.TrimSpace(opensearchSettings.AdminCert)) < 1 ||
+		len(strings.TrimSpace(opensearchSettings.PrivateKey)) < 1 ||
+		len(strings.TrimSpace(opensearchSettings.PublicKey)) < 1 {
+		return fmt.Errorf("opensearch root_ca and/or admin_key and/or admin_cert and/or public_key and/or private_key are missing. Otherwise set enable_custom_certs to false")
+	}
+
+	return checkCertValid([]keydetails{
+		{key: opensearchSettings.RootCA, certtype: ROOT_CA, svc: OPENSEARCH},
+		{key: opensearchSettings.AdminKey, certtype: ADMIN_KEY, svc: OPENSEARCH},
+		{key: opensearchSettings.AdminCert, certtype: ADMIN_CERT, svc: OPENSEARCH},
+		{key: opensearchSettings.PrivateKey, certtype: PRIVATE_KEY, svc: OPENSEARCH},
+		{key: opensearchSettings.PublicKey, certtype: PUBLIC_KEY, svc: OPENSEARCH},
+	})
+}
+
+func validateIPList(ipList []string, prefix string) error {
+	errorList := list.New()
+	for _, element := range ipList {
+		if checkIPAddress(element) != nil {
+			errorList.PushBack(prefix + " " + element + "is not valid")
+		}
+	}
+	return getSingleErrorFromList(errorList)
+}
+
+func validateCertsByIP(certsByIpList *[]CertByIP, nodeType string) error {
+	if certsByIpList == nil {
+		return nil
+	}
+	errorList := list.New()
+	for _, element := range *certsByIpList {
+		if len(strings.TrimSpace(element.IP)) < 1 ||
+			len(strings.TrimSpace(element.PrivateKey)) < 1 ||
+			len(strings.TrimSpace(element.PublicKey)) < 1 {
+			return fmt.Errorf(nodeType + "root_ca and/or public_key and/or private_key are missing in certs_by_ip. Otherwise set enable_custom_certs to false")
```

Comment:
```suggestion
			return fmt.Errorf(nodeType + "public_key and/or private_key are missing in certs_by_ip. Otherwise set enable_custom_certs to false")
```

Reply:
Updated code.